google/cloud/pubsub/internal/tracing_pull_ack_handler.cc

```diff
@@ -38,36 +38,39 @@ using Attributes =
 
 namespace {
 
-auto CreateLinks() {
+/// Create a list of links with @p span context if compiled with open telemetery
+/// ABI 2.0.
+auto CreateLinks(
+    std::shared_ptr<opentelemetry::trace::SpanContext> const& span_context) {
   using Links =
       std::vector<std::pair<opentelemetry::trace::SpanContext, Attributes>>;
 #if OPENTELEMETRY_ABI_VERSION_NO >= 2
-  auto consumer_span_context =
-      opentelemetry::trace::GetSpan(
-          opentelemetry::context::RuntimeContext::GetCurrent())
-          ->GetContext();
-  if (consumer_span_context.IsSampled() && consumer_span_context.IsValid()) {
-    return Links{{consumer_span_context, Attributes{}}};
+  if (span_context->IsSampled() && span_context->IsValid()) {
+    return Links{{*span_context, Attributes{}}};
   }
 #endif
+  (void)span_context;
   return Links{};
 }
 
+/// Adds two link attributes to the @p current span for the trace id and span id
+/// from @span.
 void MaybeAddLinkAttributes(
-    opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> const& span) {
+    opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> const&
+        current_span,
+    std::shared_ptr<opentelemetry::trace::SpanContext> const& span_context,
+    std::string const& span_name) {
```

Comment:
I should have noticed these before, but same comment:

```suggestion
    opentelemetry::trace::Span const& current_span,
    opentelemetry::trace::SpanContext const& span_context,
    std::string const& span_name) {
```

Reply:
Done